Source/WTF/wtf/WeakHashSet.h

```diff
@@ -143,7 +138,16 @@ class WeakHashSet final {
 
     bool hasNullReferences() const
     {
-        return WTF::anyOf(m_set, [] (auto& value) { return !value.get(); });
+        unsigned count = 0;
+        auto result = WTF::anyOf(m_set, [&](auto& value) {
+            ++count;
+            return !value.get();
+        });
+        if (result)
+            amortizedCleanupIfNeeded(count);
```

Comment:
I know where doing the same thing in WeakHashMap but this looks surprising to me. The caller is asking hasNullReference() but it may cause up to clean up and drop the null references. This, in itself is surprising. But even more surprising, we'll still return true after we've done the clean up?

Reply:
Maybe shouldn't do cleanup in this function? We can align the behavior of WeakHashMap to WeakHashSet instead.

Reply:
Yes, I'd prefer if we didn't (and aligned WeakHashMap).